docs/core/concepts/execution.md

```diff
@@ -100,6 +126,34 @@ with prefect.context(key='abc'):
     flow.run() # this run is successful
 ```
 
-::: warning Modifying the context
-We strongly recommend that users treat the context as read-only. Modifications can have unintended consequences.
+### Prefect-supplied context
+In addition to your own context keys, Prefect supplies context to the context object dynamically during flow runs and task runs. This context provides some standard information about the current flow or task. For example, running tasks already know about the day they are run from Prefect-provided context:
+ 
+```python{4}
+@task
+def report_start_day():
+    logger = prefect.context.get("logger")
+    logger.info(prefect.context.today)
+
+with Flow('My flow') as flow:
+	report_start_day()
+
+flow.run()
+```
+```text{5}
+[2020-03-02 22:15:58,779] INFO - prefect.FlowRunner | Beginning Flow run for 'My flow'
+[2020-03-02 22:15:58,780] INFO - prefect.FlowRunner | Starting flow run.
+[2020-03-02 22:15:58,786] INFO - prefect.TaskRunner | Task 'report_start_time': Starting task run...
+[2020-03-02 22:15:58,786] INFO - prefect.Task: report_start_day | 2020-03-02
+[2020-03-02 22:15:58,788] INFO - prefect.TaskRunner | Task 'report_start_time': finished task run for task with final state: 'Success'
+[2020-03-02 22:15:58,789] INFO - prefect.FlowRunner | Flow run SUCCESS: all reference tasks succeeded
+```
+Using this context can be useful to write time-aware tasks, such as tasks that trigger future work respective to its start time using `prefect.context.tomorrow` or processing only the prior day's data by using `prefect.context.yesterday`.
+
+::: tip What else is in context?
+For an exhaustive list of values that you can find in context, see the corresponding [API documentation](../../api/latest/utilities/context.html).
+:::
+
+::: warning Avoid modifying Prefect-supplied context
+Since Prefect uses context internally to track metadata during the flow and task run logic, modifying Prefect-supplied context keys can have unintended consequences. It is recommended to avoid overriding the key names described in the API documentation.
```

Comment:
It might be useful to add some additional nuance here: in particular, all of the timestamp related context keys can be overwritten by users and that is a supported use of context (mainly for the purposes of "backfilling").

Reply:
Updated in https://github.com/PrefectHQ/prefect/pull/2110/commits/42fb15c3114cce60b6db336f0c2edcf89a4fad26